### ISSUE TITLE
Queen-Attack: Class constructor object in template

### ISSUE THREAD

**Issue by BoDaly** [exercism/javascript#938]
The current template is:
```
constructor({
    black: [blackRow, blackColumn],
    white: [whiteRow, whiteColumn],
  } = {}) {
    throw new Error('Remove this statement and implement this function');
  }
```
This will cause a "TypeError: undefined is not iterable" when the variables are correctly replaced with values.
To remedy, the constructor should be written:
```
constructor({
    black = [blackRow, blackColumn],
    white = [whiteRow, whiteColumn],
  } = {}) {
    throw new Error('Remove this statement and implement this function');
  }
```

**Comment by SleeplessByte** [comment]
Can you show an example of this error in conjunction with the code ran? The tests are passing, so there doesn't seem to be an issue at the moment.

**Comment by SleeplessByte** [comment]
I see the example and stub have diverged. The proposed remediation doesn't solve the underlying issue.

In your example, `blackRow` will _always_ be `undefined`, and thus what it actually is:

```javascript
constructor({
  black = [undefined, undefined],
  white = [undefined, undefined],
} = {}) {
  throw new Error('Remove this statement and implement this function');
}
```

This assigns `black` and `white`, not `blackRow` and `blackColumn`.

Instead this should be written:

```javascript
class QueenAttack {
  constructor({
    black: [blackRow, blackColumn] = [],
    white: [whiteRow, whiteColumn] = [],
  } = {}) {
    throw new Error('Remove this statement and implement this function');
  }
}
```

It aims at destructuring `blackRow`, `blackColumn`, `whiteRow` and `whiteColumn`.

For example:

```javascript
new QueenAttack({ black: [3, 4] })
// => blackColumn = 4, whiteColumn = 0
```

We take PRs, if you want to make it?

**Comment by BoDaly** [comment]
I tried running the solution provided and then assigning the props to `this` as usual with constructors but this method does not recognize `black` or `white` as being defined after an input is given. (as per the test suite):
```
      const queens = new QueenAttack({ white: [2, 2] });
      expect(queens.white).toEqual([2, 2]);
```

**Comment by SleeplessByte** [comment]
Here are some options that should work. The _reason_ the stub has the destructuring is so that it's abundantly clear what is the `blackRow` and what is the `whiteColumn`. You're free to implement it however you like:

```javascript
this.black = [blackRow, blackColumn]
this.white = [whiteRow, whiteColumn]
```

or

```javascript
class QueenAttack {
  constructor(params = {}) {
    this.black = params.black
    this.white = params.white
  }
}
```

or

```javascript
class QueenAttack {
  constructor({ black, white } = {}) {
    this.params = { black, white }
  }
  
  get black() {
    return this.params.black
  }
  
  get white() {
    return this.params.white
  }
}
```

**Comment by BoDaly** [comment]
Ok I see so this method defines `blackRow` `blackColumn` `whiteRow` `whiteColumn` not `black` or `white` as props.
My inexperience got the better of me on that one, thanks!

**Comment by SleeplessByte** [comment]
You got it! That's how it works. 

When I re-ordered the track, I did expect people to know about destructuring by the time they got to this exercise, but in v3, we'll try to solve this with a _specific exercise(s) on destructuring_.

**Comment by SleeplessByte** [comment]
Closed by #939
